premailer/premailer.py

```diff
@@ -515,6 +517,17 @@ def transform(self, html=None, pretty_print=True, **kwargs):
                 out = _cdata_regex.sub(
                     lambda m: "/*<![CDATA[*/%s/*]]>*/" % m.group(1), out
                 )
+            if self.preserve_handlebar_syntax:
+                while True:
+                    x = out.split("%7B%7B", 1)
+                    if len(x) > 1:
+                        y = x[1].split("%7D%7D", 1)
+                        if len(y) > 1:
+                            out = x[0] + "{{" + unquote(y[0]) + "}}" + y[1]
+                        else:
+                            break
+                    else:
+                        break
```

Comment:
Instead of the string-magic, how about purely looking for it in attributes that have quotation marks, as a regular expression?
E.g. 
```python
re.sub(
        r'="%7B%7B(.*?)%20%7D%7D"',
        lambda match: '="{{' + unquote(match.groups()[0]) + '}}"',
        html,
    )
```
Because `etree.tostring()` will always use `"` for attributes, right?

This way, you can be a little bit more certain it will only apply this hack on what appears to be HTML tag attributes.

Reply:
This is definitely a better way of doing it considering we know the contents of the HTML tag attributes are affected. I'm not too great at regex but your implementation definitely worked. Change included here: https://github.com/peterbe/premailer/pull/252/commits/03c0db4dc242a4b66f49facf55dfbb3a3a5bc575